drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -516,6 +516,41 @@ local function component_to_endpoint(device, component_name)
   return find_default_endpoint(device, clusters.DoorLock.ID)
 end
 
+local function do_configure(driver, device)
+  local fingerprinted_devices = {
```

Comment:
A new fingerprint was added yesterday and should be added to this list: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1650

Reply:
I think Nick makes a very good point about this requiring an update to this list every time we add a WWST fingerprint. I don't think this would be very difficult, but we have received push back on this in the past due to the way it complicates the WWST process. I think if we communicate with @lelandblue and make sure this is coordinated with the WWST team, this could be an extra step that is added to the WWST process and we can maintain this list here ourselves.

However, I think there is a different way to think about issue as well. Currently, we are trying to filter out all devices that have a fingerprint because these already have assigned profiles that we do not want to change. However, the assumption we are making here is that all non-fingerprinted devices are using the `base-lock` profile, which is true in this case.

So, what if instead of filtering out fingerprinted devices, we instead just specifically targeted devices using the `base-lock` profile? This would be a different way to get the same subset of devices, and it would prevent the need to maintain a list of fingerprinted devices in the driver. 

One way we can do this is by using the `device:supports_capability()` function to determine if a device supports a given capability, and we can call this function multiple times for each of the capabilities in the `base-lock` profile to see if the device supports all the capabilities in the `base-lock` profile. If it does, we can assume that it is using the `base-lock` profile, because based on my initial review of the profiles, the `base-lock` profile does not contain capabilities that are a subset of any other profile, so therefore if a device supports all the capabilities in `base-lock`, it is definitely using the `base-lock` profile. I have done something similar to this for the `matter-sensor` driver here, where we essentially reconstructed a profile based on the capabilities that the device supports and then we append `-battery` on the end if the device also supports battery: 
https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/967/files#diff-8a122f65a4419a8fe00de96e7444d9d71df5609075dbb217531dc6d125197755R26-R36

I think we could do something similar here where we do something like this:
```
  if device:supports_capability(capabilities.lock) and
    device:supports_capability(capabilities.lockCodes) and
    device:supports_capability(capabilities.tamperAlert) and
    device:supports_capability(capabilities.battery) then
    -- we can assume device is using base-lock profile
    -- handle switching to no battery profile as needed
  end
```

Now, we know there will be changes coming to the `matter-lock` driver that will make this assumption of "non-fingerprinted devices use `base-lock`" a little more complicated, but I think we could use the same method of essentially constructing the profile from it's capabilities that is used above to handle things more granularly once we need to consider the profiles that support `lockAlarm` with the new matter lock driver.

Reply:
Good catch!

Reply:
Yeah @ctowns I think this seems like a very reasonable idea that does the same thing without needing an extra table!